### PR TITLE
Refactoring2

### PR DESCRIPTION
**Pull request by ViachaslauKabak**
refactor role

### CODE REVIEW

File: tasks/configuration/Windows.yml [hunk 1]
```diff
@@ -1,58 +1,51 @@
...
-  set_fact:
-    solr_proto: "{{ solr_ssl_configure | ternary('https', 'http') }}"
-
-- name: "Wait for Solr is UP"
-  run_once: True
+- name: Wait for Solr is UP
   win_get_url:
-    url: "{{ solr_proto }}://{{ inventory_hostname }}:{{ solr_port }}/solr/"
-    dest: "{{ solr_win_temp_dir }}\\test.html"
+    url: '{{ solr_proto }}://{{ inventory_hostname }}:{{ solr_port }}/solr/'
```
Comment (tgadiev) [inline tasks/configuration/Windows.yml]: `inventory_hostname` - should not be used directly

File: tasks/configuration/Windows.yml [hunk 2]
```diff
@@ -65,73 +58,39 @@
...
-- name: "Create new admin user via API"
-  run_once: True
-  win_command: >
-    curl --user "solr:SolrRocks" -k \
-    "{{ solr_proto }}://{{ inventory_hostname }}:{{ solr_port }}/solr/admin/authentication" \
-    -H "Content-type:application/json" \
-    -d "{"set-user": {"{{ solr_auth_user }}" : "{{ solr_auth_pass }}"} }"
+- name: Create new admin user via API
+  win_uri:
+    url: '{{ solr_proto }}://{{ inventory_hostname }}:{{ solr_port }}/solr/admin/authentication'
```
Comment (tgadiev) [inline tasks/configuration/Windows.yml]: `inventory_hostname` !

File: tasks/configuration/Windows.yml [hunk 3]
```diff
@@ -1,58 +1,51 @@
...
-  run_once: True
+- name: Wait for Solr is UP
   win_get_url:
-    url: "{{ solr_proto }}://{{ inventory_hostname }}:{{ solr_port }}/solr/"
-    dest: "{{ solr_win_temp_dir }}\\test.html"
+    url: '{{ solr_proto }}://{{ inventory_hostname }}:{{ solr_port }}/solr/'
+    dest: '{{ solr_win_temp_dir }}\test.html'
     validate_certs: False
     url_username: solr
     url_password: SolrRocks
```
Comment (tgadiev) [inline tasks/configuration/Windows.yml]: Why hardcoding credentials in role?

File: tasks/configuration/Windows.yml [hunk 4]
```diff
@@ -65,73 +58,39 @@
...
-    curl --user "solr:SolrRocks" -k \
-    "{{ solr_proto }}://{{ inventory_hostname }}:{{ solr_port }}/solr/admin/authentication" \
-    -H "Content-type:application/json" \
-    -d "{"set-user": {"{{ solr_auth_user }}" : "{{ solr_auth_pass }}"} }"
+- name: Create new admin user via API
+  win_uri:
+    url: '{{ solr_proto }}://{{ inventory_hostname }}:{{ solr_port }}/solr/admin/authentication'
+    method: POST
+    user: solr
+    password: SolrRocks
```
Comment (tgadiev) [inline tasks/configuration/Windows.yml]: Hardcoded credentials. Why?

File: tasks/configuration/Windows.yml [hunk 5]
```diff
@@ -65,73 +58,39 @@
...
-    curl --user "solr:SolrRocks" -k \
-    "{{ solr_proto }}://{{ inventory_hostname }}:{{ solr_port }}/solr/admin/authentication" \
-    -H "Content-type:application/json" \
-    -d "{"set-user": {"solr" : "{{ solr_auth_pass }}"} }"
+- name: Change default admin user password via API
+  win_uri:
+    url: '{{ solr_proto }}://{{ inventory_hostname }}:{{ solr_port }}/solr/admin/authentication'
+    method: POST
+    user: solr
+    password: SolrRocks
```
Comment (tgadiev) [inline tasks/configuration/Windows.yml]: SolrRocks???

File: tasks/configuration/Windows.yml [hunk 6]
```diff
@@ -65,73 +58,39 @@
...
-- name: "Change default admin user password via API"
-  run_once: True
-  win_command: >
-    curl --user "solr:SolrRocks" -k \
-    "{{ solr_proto }}://{{ inventory_hostname }}:{{ solr_port }}/solr/admin/authentication" \
-    -H "Content-type:application/json" \
-    -d "{"set-user": {"solr" : "{{ solr_auth_pass }}"} }"
+- name: Change default admin user password via API
+  win_uri:
+    url: '{{ solr_proto }}://{{ inventory_hostname }}:{{ solr_port }}/solr/admin/authentication'
```
Comment (tgadiev) [inline tasks/configuration/Windows.yml]: inventory_hostname !

File: tasks/ssl/Linux.yml [hunk 7]
```diff
@@ -1,101 +1,110 @@
 ---
-- name: "Check if keystore exists in role"
+- name: Check if keystore exists in role
   stat:
-    path: "{{ role_path }}/files/{{ solr_ssl_key_store_name }}"
+    path: '{{ role_path }}/files/{{ solr_ssl_key_store_name }}'
```
Comment (tgadiev) [inline tasks/ssl/Linux.yml]: Why binding to role_path?

File: tasks/ssl/Linux.yml [hunk 8]
```diff
@@ -1,101 +1,110 @@
...
+#   become_flags: '-i'
+#   become_method: sudo
 #   when:
 #     - not keystore_file.stat.exists
 #     - ansible_version.full is version('2.7', '>=')
 
-- name: "Export certificate and private key into a bundle"
-  command: >
+- name: Export certificate and private key into a bundle
+  command: >-
```
Comment (tgadiev) [inline tasks/ssl/Linux.yml]: command? why?

File: tasks/ssl/Windows.yml [hunk 9]
```diff
@@ -1,93 +1,92 @@
...
+    creates: '{{ solr_ssl_key_store_path }}\{{ solr_ssl_key_store_name }}'
   when:
     - not keystore_file.stat.exists
-    - ansible_version.full is version('2.7', '<')
 
-- name: "Get java directory"
-  win_shell: "(Get-ChildItem Env:JAVA_HOME | \
-    Select-Object Value | ft -HideTableHeaders | Out-String).Trim()"
+- name: Get java directory
+  win_shell: '(Get-ChildItem Env:JAVA_HOME | \
```
Comment (tgadiev) [inline tasks/ssl/Windows.yml]: win_shell to get folder path?

File: tasks/system/Linux.yml [hunk 10]
```diff
@@ -1,66 +1,64 @@
...
+- name: Check if Solr Service Exists
   stat:
     path: /etc/init.d/solr
   register: solr_service_stat
 
-- name: "Set solr service status"
+- name: Set solr service status
   set_fact:
-    solr_service_status: "{{ solr_service_stat.stat.exists }}"
+    solr_service_status: '{{ solr_service_stat.stat.exists }}'
```
Comment (tgadiev) [inline tasks/system/Linux.yml]: Why separate variable?

File: tasks/system/Linux.yml [hunk 11]
```diff
@@ -1,66 +1,64 @@
...
 
-- name: "Get {{ solr_checksum }} for solr  {{ solr_version }}"
+- name: 'Get {{ solr_checksum }} for solr {{ solr_version }}'
   set_fact:
-    sha_value: "{{ sha_url_item }}"
-  with_url: "{{ solr_distr_url }}.{{ solr_checksum }}"
+    sha_value: '{{ sha_url_item }}'
```
Comment (tgadiev) [inline tasks/system/Linux.yml]: ???

File: tasks/system/Linux.yml [hunk 12]
```diff
@@ -1,66 +1,64 @@
...
 
 # initial deployment of solr
-- name: "Download solr {{ solr_version }} archive"
+- name: 'Download solr {{ solr_version }} archive'
   get_url:
-    url: "{{ solr_distr_url }}"
-    dest: "{{ dest_solr_path }}.zip"
+    url: '{{ solr_distr_url }}'
+    dest: '{{ dest_solr_path }}.zip'
     checksum: '{{ solr_checksum }}:{{ sha_value.split(" ")[0] }}'
```
Comment (tgadiev) [inline tasks/system/Linux.yml]: ???

File: tasks/system/Linux.yml [hunk 13]
```diff
@@ -69,24 +67,27 @@
...
+    group: '{{ solr_group }}'
     remote_src: True
   become: True
   when: not solr_service_status
 
-- name: "Install solr as service"
-  command: /bin/bash {{ dest_solr_path }}/bin/install_solr_service.sh \
+- name: Install solr as service
+  command: >-
+    /bin/bash {{ dest_solr_path }}/bin/install_solr_service.sh
```
Comment (tgadiev) [inline tasks/system/Linux.yml]: We really need bash here?

File: tasks/system/Windows.yml [hunk 14]
```diff
@@ -1,31 +1,32 @@
 ---
-- name: "Check if Solr Service Exists"
-  win_command: "sc query {{ solr_service_name }}"
+- name: Check if Solr Service Exists
+  win_command: 'sc query {{ solr_service_name }}'
```
Comment (tgadiev) [inline tasks/system/Windows.yml]: Why do we need this?

File: tasks/system/not-supported.yml [hunk 15]
```diff
@@ -1,5 +1,6 @@
 ---
-- name: "Warn on unsupported platform"
+- name: Warn on unsupported platform
   debug:
-    msg: "This role does not support '{{ ansible_os_family }}' platform.\
-          Please contact team@lean-delivery.com"
+    msg: >-
```
Comment (tgadiev) [inline tasks/system/not-supported.yml]:
```suggestion
    msg: |
```

File: vars/Windows.yml [hunk 16]
```diff
@@ -1,12 +1,13 @@
...
-solr_ssl_key_store_path: "{{ override_solr_ssl_key_store_path|\
-  default(dest_solr_path + '\\server\\solr', True) }}"
-solr_ssl_key_store: "{{ override_solr_ssl_key_store|\
-  default(solr_ssl_key_store_path + '\\\\' + solr_ssl_key_store_name, True) }}"
-solr_ssl_trust_store: "{{ override_solr_ssl_trust_store|\
-  default(solr_ssl_key_store_path + '\\\\' + solr_ssl_key_store_name, True) }}"
+solr_ssl_key_store_path: >-
+  {{ override_solr_ssl_key_store_path | default(dest_solr_path + '\server\solr', True) }}
+solr_ssl_key_store: >-
+  {{ override_solr_ssl_key_store | default(solr_ssl_key_store_path + '\\' + solr_ssl_key_store_name, True) }}
```
Comment (tgadiev) [inline vars/Windows.yml]: Too long string. May be break?

File: vars/Windows.yml [hunk 17]
```diff
@@ -1,12 +1,13 @@
...
+solr_ssl_trust_store: >-
+  {{ override_solr_ssl_trust_store | default(solr_ssl_key_store_path + '\\' + solr_ssl_key_store_name, True) }}
```
Comment (tgadiev) [inline vars/Windows.yml]: Long string. Break?

File: tasks/system/Linux.yml (hunk 13 above)
Comment (ViachaslauKabak) [reply tasks/system/Linux.yml]: We r using the default functionality to setup service

File: tasks/system/Windows.yml (hunk 14 above)
Comment (ViachaslauKabak) [reply tasks/system/Windows.yml]: not to resetup if solr exists

File: tasks/system/Linux.yml (hunk 13 above)
Comment (ViachaslauKabak) [reply tasks/system/Linux.yml]: will use systemd to setup service

File: tasks/system/Windows.yml (hunk 14 above)
Comment (ViachaslauKabak) [reply tasks/system/Windows.yml]: will try to remove this

File: tasks/ssl/Linux.yml [hunk 18]
```diff
@@ -1,104 +1,54 @@
 ---
-- name: "Check if keystore exists in role"
+- name: Check if keystore exists in role
   stat:
-    path: "{{ role_path }}/files/{{ solr_ssl_key_store_name }}"
+    path: '{{ solr_local_keystore }}'
   delegate_to: localhost
```
[the remark this answers is not in the document]
Comment (ViachaslauKabak) [reply tasks/ssl/Linux.yml]: Keystore file is on ansible host

**Pull request merged**